### PR TITLE
Migrate xpack packetbeat

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

Migrate `x-pack packetbeat` pipeline

Example of the pipelines:
[beats-xpack-packetbeat](https://buildkite.com/elastic/beats-xpack-packetbeat/builds/771)
Examples of the affected pipelines:
[beats-libbeat](https://buildkite.com/elastic/beats-libbeat/builds/2044)
[beats-metricbeat](https://buildkite.com/elastic/beats-metricbeat/builds/3258)
[beats-packetbeat](https://buildkite.com/elastic/beats-packetbeat/builds/2051)
[beats-winlogbeat](https://buildkite.com/elastic/beats-winlogbeat/builds/1207)
[beats-xpack-libbeat](https://buildkite.com/elastic/beats-xpack-libbeat/builds/814)
[beats-xpack-metricbeat](https://buildkite.com/elastic/beats-xpack-metricbeat/builds/782)

## Related issues
https://github.com/elastic/ingest-dev/issues/1693
https://github.com/elastic/beats/issues/38142 - related to commented-out Windows system tests

### CODE REVIEW

**Comment by sharbuz** [comment]
This PR was made from https://github.com/elastic/beats/pull/38081, please don't touch it till [the PR](https://github.com/elastic/beats/pull/38081) is merged.
@dliappis @v1v

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
 
 steps:
+
```
Comment (v1v) [inline .buildkite/pipeline.yml:4]: what's the reason for this change?

File: .buildkite/scripts/win_unit_tests.ps1 [hunk 2]
```diff
@@ -120,6 +124,14 @@ function withNmap($version) {
     }
     Start-Process -FilePath $nmapDownloadPath -ArgumentList "/S" -Wait
 }
+function google_cloud_auth {
+    $tempFileName = "google-cloud-credentials.json"
+    $secretFileLocation = Join-Path $env:TEMP $tempFileName
```
Comment (v1v) [inline .buildkite/scripts/win_unit_tests.ps1:129]: Shall this file be deleted after being used?

File: .buildkite/x-pack/pipeline.xpack.metricbeat.yml [hunk 3]
```diff
@@ -1,5 +1,60 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+name: "beats-xpack-metricbeat"
```
[the remark this answers is not in the document]
Comment (sharbuz) [reply .buildkite/x-pack/pipeline.xpack.metricbeat.yml:2]: please take a look here: https://github.com/elastic/beats/pull/38135#issuecomment-1969220857 :)
I hoped to merge https://github.com/elastic/beats/pull/38081 faster :(

File: .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh [hunk 4]
```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
+
+source .buildkite/scripts/common.sh
+
+set -euo pipefail
+
+pipelineName="pipeline.xpack-metricbeat-dynamic.yml"
```
Comment (v1v) [inline .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh:7]: name is not packetbeat
```suggestion
pipelineName="pipeline.xpack-packetbeat-dynamic.yml"
```

File: .buildkite/scripts/win_unit_tests.ps1 (hunk 2 above)
Comment (v1v) [reply .buildkite/scripts/win_unit_tests.ps1:129]: This discussion has not been answered

File: .buildkite/scripts/setenv.sh [hunk 5]
```diff
@@ -11,6 +11,8 @@ SETUP_WIN_PYTHON_VERSION="3.11.0"
 NMAP_WIN_VERSION="7.12"           # Earlier versions of NMap provide WinPcap (the winpcap packages don't install nicely because they pop-up a UI)
 GO_VERSION=$(cat .go-version)
 ASDF_MAGE_VERSION="1.15.0"
+PACKAGING_PLATFORMS="+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
+PACKAGING_ARM_PLATFORMS="linux/arm64"
```
Comment (v1v) [inline .buildkite/scripts/setenv.sh:15]: This is duplicated, see L17/L18

```suggestion
```

File: .buildkite/scripts/win_unit_tests.ps1 [hunk 6]
```diff
@@ -142,10 +154,19 @@ $env:MAGEFILE_CACHE = $magefile
...
-} else {
-    mage build unitTest
+if ($testType -eq "unittest") {
+    if ($env:BUILDKITE_PIPELINE_SLUG -eq "beats-xpack-libbeat") {
+        mage -w reader/etw build goUnitTest
+    } else {
+        mage build unitTest
+    }
+}
+elseif ($testType -eq "systemtest") {
```
Comment (v1v) [inline .buildkite/scripts/win_unit_tests.ps1:175]: The name of the file `.buildkite/scripts/win_unit_tests.ps1` is specific about `unit_tests` while it also runs ` mage systemTest`.

What do you think about this proposal that can avoid maintaining the `.buildkite/scripts/win_unit_tests.ps1` but the pipeline itself?


rename the file for `buildkite/scripts/win_mage.ps1`, it's more generic and could accept different mage goals.

`testType` to be renamed for `mageGoals` and then simplify the script with the below snippet:

```
if ($mageGoals -eq "systemTest") {
    google_cloud_auth
}

mage $mageGoals
```

Change the pipeline stages with:

```yaml
# .buildkite/scripts/generate_xpack_packetbeat_pipeline.sh
      - label: ":windows: Windows Unit Tests - {{matrix.image}}"
        command: ".buildkite/scripts/win_mage.ps1 build unitTest"
     ...
      ##  TODO: uncomment when the issue https://github.com/elastic/beats/issues/38142 is solved
      # - label: ":windows: Windows 10 System Tests"
      #   key: "extended-win-10-system-tests"
      #.  command: ".buildkite/scripts/win_mage.ps1 systemTest"

```

```yaml
# .buildkite/scripts/generate_xpack_ libbeat_pipeline.sh
      - label: ":windows: Windows 11 Unit Tests"
        command: ".buildkite/scripts/win_mage.ps1 -w reader/etw build goUnitTest"

```

File: .buildkite/scripts/setenv.sh (hunk 5 above)
Comment (sharbuz) [reply .buildkite/scripts/setenv.sh]: fixed

File: .buildkite/scripts/win_unit_tests.ps1 (hunk 6 above)
Comment (sharbuz) [reply .buildkite/scripts/win_unit_tests.ps1:175]: I agree with you that it could be done as you suggest. There is only one remark. We gonna decline to use Windows scripts soon (I hope in 1 week). If it's possible to avoid additional changes right now, let's do that. @v1v WDYT?

Comment (v1v) [reply .buildkite/scripts/win_unit_tests.ps1:175]: Do you mean the file `.buildkite/scripts/win_unit_tests.ps1` will not be used anymore soon? I still think the name of the file is not correct for the nature of what it does, so `.buildkite/scripts/win_tests.ps1` could be more meaningful.

File: .buildkite/scripts/win_unit_tests.ps1 (hunk 2 above)
Comment (sharbuz) [reply .buildkite/scripts/win_unit_tests.ps1:129]: I've added cleanup [here](https://github.com/elastic/beats/pull/38135/commits/d02580b612334c9b2875a1d3c3a8bda71216ba83), [there is](https://buildkite.com/elastic/beats-xpack-packetbeat/builds/796#018e378a-9aef-49ac-b056-6328a011d99f/101-309) the result, please retake a look.

File: .buildkite/scripts/win_unit_tests.ps1 (hunk 6 above)
Comment (sharbuz) [reply .buildkite/scripts/win_unit_tests.ps1:173]: Correct.
We are going to use our own BK Windows images and @dliappis is working on it.
When it's ready we are going to use command steps with 1-3 commands without using PowerShell scripts.

Comment (sharbuz) [reply .buildkite/scripts/win_unit_tests.ps1:173]: Renaming this script requires a lot of changes in the already prepared dynamic pipelines.

Comment (v1v) [reply .buildkite/scripts/win_unit_tests.ps1:173]: I'll then wait for @dliappis 's feedback; if he has no concerns, I'm happy with this.

Comment (dliappis) [reply .buildkite/scripts/win_unit_tests.ps1:173]: Thanks @v1v for the great feedback (as usual!) The custom Windows images are now ready, so I am good with letting this as is and get rid of many custom scripts and snippets, including the one discussed here, ASAP.

File: x-pack/packetbeat/magefile.go [hunk 7]
```diff
@@ -172,6 +172,13 @@ func SystemTest(ctx context.Context) error {
 	return devtools.GoTest(ctx, args)
 }
 
+func getBucketName() string {
```
Comment (dliappis) [inline x-pack/packetbeat/magefile.go:175]: @sharbuz if we haven't it yet, let's create a post migration ticket in the usual repo and link it to https://github.com/elastic/ingest-dev/issues/2794 to remind us that the conditional behavior here should be removed once Jenkins has been sunset (I realize that it should be possible to run the tests locally by devs too but I believe they have access to it already).

**Review by dliappis — APPROVED** [review]
LGTM thanks all for the great work and review discussions!

**Comment by jlind23** [comment]
Hey @nfritts @norrietaylor, your team approval is required here even though this is purely a CI change that we own.
Can we please get your thumbs up?

File: x-pack/packetbeat/magefile.go (hunk 7 above)
Comment (sharbuz) [reply x-pack/packetbeat/magefile.go:175]: done.
https://github.com/elastic/ingest-dev/issues/3006

**Review by mjwolf — APPROVED** [review]

**Comment by jlind23** [comment]
Force merging this as it is blocked due to unrelated problems, no MacOS workers are available to execute the job
